src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java

```diff
@@ -1023,4 +1015,31 @@ public String getCastScriptStatement(String name, String castType, List<KVValue>
                 );
         }
     }
+
+    public static Schema.Type getOrderByFieldType(Field field) {
+        String functionName = ((ScriptMethodField) field).getFunctionName().toLowerCase();
+        if (functionName.equals("cast")) {
+            String castType = ((SQLCastExpr) field.getExpression()).getDataType().getName();
+            return getCastFunctionReturnType(castType);
+        }
+
+        if (numberOperators.contains(functionName) || mathConstants.contains(functionName)
+                || trigFunctions.contains(functionName) || binaryOperators.contains(functionName)) {
+            return Schema.Type.DOUBLE;
```

Comment:
A reminder: the number operators do not always return `DOUBLE`, for example, `abs(integer)` should return `INTEGER`, and `sign(...)` should always return an integer type.

Reply:
This is just for the `ORDER BY` clause- the `scriptSortType` will only return either `STRING` or `NUMBER`, so returning `DOUBLE` for every numeric field in the `ORDER BY` will not affect the field information of the fields in `SELECT`